include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -166,9 +273,117 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
     }
 };
 
+template <typename... _Name>
+class _find_split_points_kernel_on_mid_diagonal_uint32_t;
+
+template <typename... _Name>
+class _find_split_points_kernel_on_mid_diagonal_uint64_t;
+
+template <typename _IdType, typename _CustomName, typename... _Name>
+struct __parallel_merge_submitter_large<_IdType, _CustomName, __internal::__optional_kernel_name<_Name...>>
+{
+    template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3, typename _Compare>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3, _Compare __comp) const
+    {
+        const _IdType __n1 = __rng1.size();
+        const _IdType __n2 = __rng2.size();
+        const _IdType __n = __n1 + __n2;
+
+        assert(__n1 > 0 || __n2 > 0);
+
+        _PRINT_INFO_IN_DEBUG_MODE(__exec);
+
+        using _FindSplitPointsKernelOnMidDiagonal = std::conditional_t<
+            std::is_same_v<_IdType, std::uint32_t>,
+            oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
+                _find_split_points_kernel_on_mid_diagonal_uint32_t, _CustomName, _Range1, _Range2, _IdType, _Compare>,
+            oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
+                _find_split_points_kernel_on_mid_diagonal_uint64_t, _CustomName, _Range1, _Range2, _IdType, _Compare>>;
+
+        // Empirical number of values to process per work-item
+        const std::uint8_t __chunk = __exec.queue().get_device().is_cpu() ? 128 : 4;
+
+        const _IdType __steps = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __chunk);
+        const _IdType __base_diag_count = 1'024 * 32;
+        const _IdType __base_diag_part = oneapi::dpl::__internal::__dpl_ceiling_div(__steps, __base_diag_count);
+
+        using _split_point_t = std::pair<_IdType, _IdType>;
+
+        using __result_and_scratch_storage_t = __result_and_scratch_storage<_ExecutionPolicy, _split_point_t>;
+        __result_and_scratch_storage_t __result_and_scratch{__exec, 0, __base_diag_count + 1};
+
+        sycl::event __event = __exec.queue().submit([&](sycl::handler& __cgh) {
+            oneapi::dpl::__ranges::__require_access(__cgh, __rng1, __rng2);
+            auto __scratch_acc = __result_and_scratch.template __get_scratch_acc<sycl::access_mode::write>(
+                __cgh, __dpl_sycl::__no_init{});
+
+            __cgh.parallel_for<_FindSplitPointsKernelOnMidDiagonal>(
```

Comment:
@rarutyun Compile error is here: 
https://github.com/oneapi-src/oneDPL/actions/runs/11722920053/job/32653481992?pr=1933
"C++
D:\a\oneDPL\oneDPL\include\oneapi\dpl\pstl\hetero\dpcpp\parallel_backend_sycl_merge.h(322,64): error: definition with same mangled name '_ZTSN6oneapi3dpl20__par_backend_hetero10__internal11__compositeINS1_50_find_split_points_kernel_on_mid_diagonal_uint32_tIJEEEJLc95ELc90ELc84ELc83ELc78ELc54ELc111ELc110ELc101ELc97ELc112ELc105ELc51ELc100ELc112ELc108ELc50ELc48ELc95ELc95ELc112ELc97ELc114ELc95ELc98ELc97ELc99ELc107ELc101ELc110ELc100ELc95ELc104ELc101ELc116ELc101ELc114ELc111ELc53ELc48ELc95ELc102ELc105ELc110ELc100ELc95ELc115ELc112ELc108ELc105ELc116ELc95ELc112ELc111ELc105ELc110ELc116ELc115ELc95ELc107ELc101ELc114ELc110ELc101ELc108ELc95ELc111ELc110ELc95ELc109ELc105ELc100ELc95ELc100ELc105ELc97ELc103ELc111ELc110ELc97ELc108ELc95ELc117ELc105ELc110ELc116ELc51ELc50ELc95ELc116ELc73ELc74ELc78ELc83ELc48ELc95ELc57ELc101ELc120ELc101ELc99ELc117ELc116ELc105ELc111ELc110ELc53ELc95ELc95ELc100ELc112ELc108ELc49ELc55ELc68ELc101ELc102ELc97ELc117ELc108ELc116ELc75ELc101ELc114ELc110ELc101ELc108ELc78ELc97ELc109ELc101ELc69ELc78ELc83ELc48ELc95ELc56ELc95ELc95ELc114ELc97ELc110ELc103ELc101ELc115ELc49ELc48ELc103ELc117ELc97ELc114ELc100ELc95ELc118ELc105ELc101ELc119ELc73ELc80ELc105ELc69ELc69ELc83ELc57ELc95ELc106ELc78ELc83ELc48ELc95ELc49ELc48ELc95ELc95ELc105ELc110ELc116ELc101ELc114ELc110ELc97ELc108ELc49ELc49ELc95ELc95ELc112ELc115ELc116ELc108ELc95ELc108ELc101ELc115ELc115ELc69ELc69ELc69ELc69EEEE' as another definition
  322 |                 sycl::range</*dim=*/1>(__base_diag_count + 1), [=](sycl::item</*dim=*/1> __item_id)
      |                                                                ^
"